### PR TITLE
Public jbrowse

### LINKED ISSUE #1640: Logged out Apollo/JBrowse should support reverse complement
- [ ] when loading refseq double forward and reverse so we can quickly go back and forth

### PR DESCRIPTION
**Pull request by nathandunn**
Fixes #1640.

### CODE REVIEW

**Comment by nathandunn** [comment]
- [ ] when it flips it should flip to the correct location 
- [ ] is not showing proper stuff in the search box once flipped
- [ ] should be able to handle subsequent flips
- [ ] clean up 

I think that most of it has to do with encoding.

**Comment by nathandunn** [comment]
Need to assess the difference in change between generated URLs.

**Comment by nathandunn** [comment]
Working encoding:

```

http://localhost:8080/apollo/7247174766691461199310/jbrowse/index.html?loc=%7B%22sequenceList%22%3A%5B%7B%22name%22%3A%22Group11.18%22%2C%22start%22%3A0%2C%22end%22%3A4736299%2C%22reverse%22%3Atrue%2C%22organism%22%3A%22Honeybee%22%2C%22location%22%3A%5B%7B%22fmin%22%3A0%2C%22fmax%22%3A4736299%7D%5D%7D%5D%7D%3A1..217636&clientToken=7247174766691461199310&highlight=&tracklist=0&locationBox=none

{"sequenceList":[{"name":"Group11.18","start":0,"end":4736299,"reverse":true,"organism":"Honeybee","location":[{"fmin":0,"fmax":4736299}]}]}:1..217636


http://localhost:8080/apollo/21/jbrowse/index.html?loc=%7B%22sequenceList%22%3A%5B%7B%22seqChunkSize%22%3A20000%2C%22length%22%3A4736299%2C%22name%22%3A%22Group11.18%22%2C%22start%22%3A0%2C%22end%22%3A4736299%2C%22reverse%22%3Atrue%7D%5D%7D%3A4518661..4736299&tracks=DNA%2CAnnotations%2COfficial%20Gene%20Set%20v3.2&highlight=

{"sequenceList":[{"seqChunkSize":20000,"length":4736299,"name":"Group11.18","start":0,"end":4736299,"reverse":true}]}:4518661..4736299
```

vs non-working:

```

http://localhost:8080/apollo/21/jbrowse/index.html?loc=%7B%22sequenceList%22%3A%5B%7B%22seqChunkSize%22%3A20000%2C%22length%22%3A4736299%2C%22name%22%3A%22Group11.18%22%2C%22start%22%3A0%2C%22end%22%3A4736299%2C%22reverse%22%3Atrue%7D%5D%7D%3A4518661..4736299&tracks=DNA%2CAnnotations%2COfficial%20Gene%20Set%20v3.2&highlight=

{"sequenceList":[{"seqChunkSize":20000,"length":4736299,"name":"Group11.18","start":0,"end":4736299,"reverse":true}]}:4518661..4736299
```

**Comment by nathandunn** [comment]
This search box is not being ingested correctly: 

    {"sequenceList":[{"seqChunkSize":20000,"length":2027,"name":"GroupUn780","start":0,"end":2027,"reverse":true}]}:496..901:971..1376 (407 b)

**Comment by nathandunn** [comment]
@deepakunni3  I think this is actually ready for review!

**Comment by deepakunni3** [comment]
Ran into an issue.

To reproduce,

1. Go to public mode for an organism
screen shot 2017-07-14 at 6 18 51 pm

2. Do a reverse complement
screen shot 2017-07-14 at 6 19 05 pm

3. Now revert back to the original state
screen shot 2017-07-14 at 6 19 20 pm

Notice how the view is on a separate region of the genome and the scaffold still hasn't reverted back to the original state. Also note the white band right below the navigation buttons.

4. Hit refresh
screen shot 2017-07-14 at 6 19 36 pm

Now the `Group1.10` is gone and so has the textbox.

Console gives the following stack trace:
```
SyntaxError: Unexpected token : in JSON at position 268 SyntaxError: Unexpected token : in JSON at position 268
    at JSON.parse (<anonymous>)
    at Object.findReferenceSequence (http://localhost:8098/apollo/19/jbrowse/src/JBrowse/Browser.js:2441:32)
    at Object.<anonymous> (http://localhost:8098/apollo/19/jbrowse/src/JBrowse/Browser.js:2474:39)
    at http://localhost:8098/apollo/19/jbrowse/src/dojo/_base/lang.js:375:55
    at http://localhost:8098/apollo/19/jbrowse/src/JBrowse/Browser.js:1979:29
    at signalListener (http://localhost:8098/apollo/19/jbrowse/src/dojo/Deferred.js:37:21)
    at Deferred.then.promise.then (http://localhost:8098/apollo/19/jbrowse/src/dojo/Deferred.js:258:5)
    at Object.afterMilestone (http://localhost:8098/apollo/19/jbrowse/src/JBrowse/Browser.js:1977:10)
    at Object.navigateToLocation (http://localhost:8098/apollo/19/jbrowse/src/JBrowse/Browser.js:2470:10)
    at Object.callLocation (http://localhost:8098/apollo/19/jbrowse/src/JBrowse/Browser.js:2396:15) SyntaxError: Unexpected token : in JSON at position 268
    at JSON.parse (<anonymous>)
    at Object.findReferenceSequence (Browser.js:2441)
    at Object.<anonymous> (Browser.js:2474)
    at lang.js:375
    at Browser.js:1979
    at signalListener (Deferred.js:37)
    at Deferred.then.promise.then (Deferred.js:258)
    at Object.afterMilestone (Browser.js:1977)
    at Object.navigateToLocation (Browser.js:2470)
    at Object.callLocation (Browser.js:2396)

dojo.js:1250 Uncaught SyntaxError: Unexpected token : in JSON at position 268
    at JSON.parse (<anonymous>)
    at Object.getUniqueTrackName (AnnotTrack.js:5457)
    at Object.getPermission (AnnotTrack.js:5391)
    at Object.setViewInfo (AnnotTrack.js:194)
    at Object.<anonymous> (GenomeView.js:2264)
    at lang.js:375
    at GenomeView.js:2290
    at Object.<anonymous> (Browser.js:1766)
    at lang.js:375
    at runFactory (dojo.js:1104)
getUniqueTrackName @ AnnotTrack.js:5457
getPermission @ AnnotTrack.js:5391
setViewInfo @ AnnotTrack.js:194
(anonymous) @ GenomeView.js:2264
(anonymous) @ lang.js:375
(anonymous) @ GenomeView.js:2290
(anonymous) @ Browser.js:1766
(anonymous) @ lang.js:375
runFactory @ dojo.js:1104
execModule @ dojo.js:1232
(anonymous) @ dojo.js:1270
guardCheckComplete @ dojo.js:1247
checkComplete @ dojo.js:1265
onLoadCallback @ dojo.js:1431
onLoad @ dojo.js:1684

WebApolloCanvasFeatures.js:40 WA config <body class=​"colorCds Apollo tundra" jserror=​"Uncaught SyntaxError:​ Unexpected token :​ in JSON at position 268">​…​</body>​
VM9279:1 Uncaught SyntaxError: Unexpected token : in JSON at position 268
    at JSON.parse (<anonymous>)
    at Object.getUniqueTrackName (AnnotTrack.js:5457)
    at Object.loadTranslationTable (SequenceTrack.js:171)
    at Object.setAnnotTrack (SequenceTrack.js:1138)
    at Object.<anonymous> (SequenceTrack.js:122)
    at lang.js:375
```

**Comment by nathandunn** [comment]
K thanks

**Comment by nathandunn** [comment]
- [ ] canvas track error (volvox) logged out or in
- [ ] in logged out mode (for yeast) it fails when changing (works fine for Honeybee and volvox)
- [ ] on load, we need to either nudge or leave some space at the end because of excess space
- [ ] for volvox, shows UCA and extra space

== volvox error == 

    http://localhost:8080/apollo/20557/jbrowse/index.html?loc=%7B%22sequenceList%22%3A%5B%7B%22seqChunkSize%22%3A20000%2C%22start%22%3A0%2C%22length%22%3A50001%2C%22name%22%3A%22ctgA%22%2C%22end%22%3A50001%2C%22sequenceList%22%3A%5B%7B%22seqChunkSize%22%3A20000%2C%22color%22%3A%22green%22%2C%22offset%22%3A0%2C%22start%22%3A0%2C%22name%22%3A%22ctgA%22%2C%22length%22%3A50001%2C%22end%22%3A50001%2C%22reverse%22%3Atrue%2C%22sequenceList%22%3A%5B%7B%22seqChunkSize%22%3A20000%2C%22start%22%3A0%2C%22name%22%3A%22ctgA%22%2C%22length%22%3A50001%2C%22end%22%3A50001%7D%5D%2C%22order%22%3A0%7D%5D%2C%22reverse%22%3Atrue%7D%5D%7D%3A46299..49009%3A993..3692%3A1..50001%3A1..50001&tracks=DNA%2CAnnotations%2CGenes%2CCDS%2CTranscript&highlight=

```
B: {"sequenceList":[{"seqChunkSize":20000,"start":0,"length":50001,"name":"ctgA","end":50001,"sequenceList":[{"seqChunkSize":20000,"color":"green","offset":0,"start":0,"name":"ctgA","length":50001,"end":50001,"reverse":true,"sequenceList":[{"seqChunkSize":20000,"start":0,"name":"ctgA","length":50001,"end":50001}],"order":0}],"reverse":true}]}:46299..49009
:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:1981 SyntaxError: Unexpected token : in JSON at position 341 SyntaxError: Unexpected token : in JSON at position 341
    at JSON.parse (<anonymous>)
    at Object.findReferenceSequence (http://localhost:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2426:29)
    at Object.<anonymous> (http://localhost:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2474:39)
    at http://localhost:8080/apollo/20557/jbrowse/src/dojo/_base/lang.js:375:55
    at http://localhost:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:1979:29
    at signalListener (http://localhost:8080/apollo/20557/jbrowse/src/dojo/Deferred.js:37:21)
    at Deferred.then.promise.then (http://localhost:8080/apollo/20557/jbrowse/src/dojo/Deferred.js:258:5)
    at Object.afterMilestone (http://localhost:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:1977:10)
    at Object.navigateToLocation (http://localhost:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2470:10)
    at Object.callLocation (http://localhost:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2396:15) SyntaxError: Unexpected token : in JSON at position 341
    at JSON.parse (<anonymous>)
    at Object.findReferenceSequence (:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2426)
    at Object.<anonymous> (:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2474)
    at :8080/apollo/20557/jbrowse/src/dojo/_base/lang.js:375
    at :8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:1979
    at signalListener (:8080/apollo/20557/jbrowse/src/dojo/Deferred.js:37)
    at Deferred.then.promise.then (:8080/apollo/20557/jbrowse/src/dojo/Deferred.js:258)
    at Object.afterMilestone (:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:1977)
    at Object.navigateToLocation (:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2470)
    at Object.callLocation (:8080/apollo/20557/jbrowse/src/JBrowse/Browser.js:2396)
```

and 

screen shot 2017-07-14 at 8 35 43 pm

and

```

:8080/apollo/20557/jbrowse/data/tracks/CDS/%7B%22sequenceList%22:[%7B%22seq…],%22reverse%22:true%7D]%7D:46299..49009:993..3692:1..50001/trackData.json Failed to load resource: the server responded with a status of 500 (Internal Server Error)
9:8080/apollo/20557/jbrowse/src/JBrowse/View/Track/BlockBased.js:456
```

=== spacing error ===

screen shot 2017-07-14 at 8 34 23 pm


=== yeast error ===

```
http://localhost:8080/apollo/20557/jbrowse/index.html?loc=chrII%3A771675..772934&tracks=DNA%2CAnnotations&highlight=
```

to

```
http://localhost:8080/apollo/20557/jbrowse/index.html?loc=%7B%22sequenceList%22%3A%5B%7B%22seqChunkSize%22%3A20000%2C%22organism%22%3A%22org.bbop.apollo.Organism%20%3A%2020571%22%2C%22organismId%22%3A20571%2C%22sequenceChunks%22%3A%22%5B%5D%22%2C%22featureLocations%22%3A%22%5B%5D%22%2C%22name%22%3A%22chrII%22%2C%22start%22%3A0%2C%22length%22%3A813178%2C%22end%22%3A813178%2C%22reverse%22%3Atrue%7D%5D%7D%3A771674..772934&tracks=DNA%2CAnnotations&highlight=
```

=== Canvas Error track (when logged in)

screen shot 2017-07-14 at 8 27 33 pm

screen shot 2017-07-14 at 8 28 50 pm


```
putative sequence path name /opt/apollo/volvox/tracks/CDS/{"sequenceList":[{"name":"ctgA","start":0,"end":50001,"reverse":false,"organism":"volvox","location":[{"fmin":0,"fmax":50001}]}]}:0..50001/trackData.json -> {"sequenceList":[{"name":"ctgA","start":0,"end":50001,"reverse":false,"organism":"volvox","location":[{"fmin":0,"fmax":50001}]}]}:0..50001 
putative sequence array [{"organism":"volvox","name":"ctgA","start":0,"end":50001,"location":[{"fmin":0,"fmax":50001}],"reverse":false}]
found a projection 1
| Error 2017-07-14 20:27:31,778 [http-bio-8080-exec-3] ERROR errors.GrailsExceptionResolver  - JSONException occurred when processing request: [GET] /apollo/1769332111325639066443668980/jbrowse/data/tracks/CDS/%7B%22sequenceList%22:[%7B%22name%22:%22ctgA%22,%22start%22:0,%22end%22:50001,%22reverse%22:false,%22organism%22:%22volvox%22,%22location%22:[%7B%22fmin%22:0,%22fmax%22:50001%7D]%7D]%7D:0..50001/trackData.json
JSONArray[11] is not a JSONObject.. Stacktrace follows:
Message: JSONArray[11] is not a JSONObject.
    Line | Method
->>  588 | sanitizeCoordinateArray in org.bbop.apollo.TrackService
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|    522 | $tt__loadTrackData      in     ''
|   1023 | $tt__projectTrackData . in     ''
|    337 | data                    in org.bbop.apollo.JbrowseController
|    198 | doFilter . . . . . . .  in grails.plugin.cache.web.filter.PageFragmentCachingFilter
|     63 | doFilter                in grails.plugin.cache.web.filter.AbstractFilter
|    449 | executeChain . . . . .  in org.apache.shiro.web.servlet.AbstractShiroFilter
|    365 | call                    in org.apache.shiro.web.servlet.AbstractShiroFilter$1
|     90 | doCall . . . . . . . .  in org.apache.shiro.subject.support.SubjectCallable
|     83 | call                    in     ''
|    383 | execute . . . . . . . . in org.apache.shiro.subject.support.DelegatingSubject
|    362 | doFilterInternal        in org.apache.shiro.web.servlet.AbstractShiroFilter
|    125 | doFilter . . . . . . .  in org.apache.shiro.web.servlet.OncePerRequestFilter
|   1142 | runWorker               in java.util.concurrent.ThreadPoolExecutor
|    617 | run . . . . . . . . . . in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run                     in java.lang.Thread
current user? org.bbop.apollo.User : 20
getting this token 176933211132563906644366898
```

**Comment by nathandunn** [comment]
- [ ] when viewing BigWig files get other errors (below) having to do with getting prefernce from the database instead of Token / Organism ID
- [ ] when flipping back and forth does not seem to be pruning properly


===

BigWig error
```
15068:1..75085:1..75085 - parameters:
basesPerSpan: 37.80715005035246
urlTemplate: bigwig/nurse.bw
start: 85067
scale: 0.026450023306918827
end: 94520
User does not have permission for any organisms.. Stacktrace follows:
Message: User does not have permission for any organisms.
    Line | Method
->>  479 | $tt__getCurrentOrganismPreferenceInDB in org.bbop.apollo.PreferenceService
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|    402 | $tt__getCurrentOrganismPreference     in     ''
|     45 | features . . . . . . . . . . . . . .  in org.bbop.apollo.BigwigController
|    198 | doFilter                              in grails.plugin.cache.web.filter.PageFragmentCachingFilter
|     63 | doFilter . . . . . . . . . . . . . .  in grails.plugin.cache.web.filter.AbstractFilter
|    449 | executeChain                          in org.apache.shiro.web.servlet.AbstractShiroFilter
|    365 | call . . . . . . . . . . . . . . . .  in org.apache.shiro.web.servlet.AbstractShiroFilter$1
|     90 | doCall                                in org.apache.shiro.subject.support.SubjectCallable
|     83 | call . . . . . . . . . . . . . . . .  in     ''
|    383 | execute                               in org.apache.shiro.subject.support.DelegatingSubject
|    362 | doFilterInternal . . . . . . . . . .  in org.apache.shiro.web.servlet.AbstractShiroFilter
|    125 | doFilter                              in org.apache.shiro.web.servlet.OncePerRequestFilter
|   1142 | runWorker . . . . . . . . . . . . . . in java.util.concurrent.ThreadPoolExecutor
|    617 | run                                   in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run . . . . . . . . . . . . . . . . . in java.lang.Thread
current user? null
data directory /opt/apollo/
```

**Comment by nathandunn** [comment]
The immediate problem is that it seems to continue to build up sequence irregularities.

**Comment by nathandunn** [comment]
- init condition:  ```Group11.4%3A8974..20612```
- first reverse (correct):
```
 %7B"sequenceList"%3A%5B%7B"seqChunkSize"%3A20000%2C"length"%3A75085%2C"name"%3A"Group11.4"%2C"start"%3A0%2C"end"%3A75085%2C"reverse"%3Atrue%7D%5D%7D%3A8973..20612%3A54472..66111
```
- second reverse (BAD): 

```
%7B"sequenceList"%3A%5B%7B"seqChunkSize"%3A20000%2C"start"%3A0%2C"length"%3A75085%2C"name"%3A"Group11.4"%2C"end"%3A75085%2C"sequenceList"%3A%5B%7B"seqChunkSize"%3A20000%2C"color"%3A"green"%2C"offset"%3A0%2C"length"%3A75085%2C"name"%3A"Group11.4"%2C"start"%3A0%2C"end"%3A75085%2C"reverse"%3Atrue%2C"order"%3A0%7D%5D%2C"reverse"%3Atrue%7D%5D%7D%3A54471..66111%3A8977..20617
```

**Comment by nathandunn** [comment]
This works for both Volvox and Yeast, but I noticed some problems with a few of the track types when being reversed in Volvox /ctgA:

- [ ] GFF3 in-memory (not showing up at all)
- [ ] GTF in-memory (not showing up at all)
- [ ] .wig track (500 errors for "not found")
- [ ] bigwig tracks when reversed
- [ ] bigwig tracks at all (for Honeybee) 

The Honeybee tracks are because its not passing in the organism to the BigWigController.

**Comment by nathandunn** [comment]
Moved filetype bugs to #1703, #1704, and #1705, which is separate from this.

**Comment by nathandunn** [comment]
@deepakunni3  This is ready for review.  I tested logged out and logged in as well as the navigation and everything seemed to work except as noted above for the different filetypes (which I have to fix anyway).

File: client/apollo/js/main.js [hunk 1]
```diff
@@ -931,12 +931,7 @@ return declare( [JBPlugin, HelpMixin,Evented],
                     locationBox.set('value',locationVal,false);
                 }
                 else{
-                    var locationBoxString = Util.assembleLocStringWithLength( currRegion );
-                    if(currRegion.ref.startsWith("{")){
-                        sequenceString = locationBoxString.substring(0,locationBoxString.lastIndexOf("}")+1);
-                        sequenceObject = JSON.parse(sequenceString).sequenceList[0];
-                        locationBoxString = sequenceObject.name + locationBoxString.substr(locationBoxString.lastIndexOf(":"));
-                    }
+                    var locationBoxString = Util.assembleProjectedString(currRegion);
```
Comment (deepakunni3) [inline client/apollo/js/main.js]: There is no function `assembleProjectedString` in `JBrowse/Util.js`

I get this error on the console when clicking anywhere on the browser:
```
main.js:961 Uncaught TypeError: Util.assembleProjectedString is not a function
    at main.js:961
    at Evented.<anonymous> (lang.js:375)
    at Evented.dispatcher [as on/jbrowse/v1/n/navigate] (aspect.js:101)
    at Function.on.emit (on.js:309)
    at Function.on.emit (on.js:350)
    at Evented.emit (Evented.js:32)
    at Object.publish (topic.js:25)
    at Object.publish (Browser.js:1877)
    at Object.onCoarseMove (Browser.js:2896)
    at Object.advice (aspect.js:114)
```

**Comment by deepakunni3** [comment]
@nathandunn Still seeing the white band right below the navigation buttons.
screen shot 2017-07-18 at 10 44 47 am

**Comment by nathandunn** [comment]
K.  I can take a look at that.  Let me know how everything else looks. 

Nathan

File: client/apollo/js/main.js (hunk 1 above)
Comment (nathandunn) [inline client/apollo/js/main.js]: @deepakunni3  Sorry, you need to pull your jbrowse.  

https://github.com/nathandunn/jbrowse/

branch  projection_v2_server

**Comment by deepakunni3** [comment]
I know I am supposed to be testing the public jbrowse mode but I am running into a few issues when logged in.

While browsing in logged-in mode, I end up with the following error:
screen shot 2017-07-18 at 11 14 54 am

And on the console:
```
instrumentation.js:20 Error: SyntaxError: Unexpected token < in JSON at position 0
    at NCList.js:181
    at signalListener (Deferred.js:37)
    at signalWaiting (Deferred.js:28)
    at Deferred.reject.reject (Deferred.js:219)
    at signalDeferred (Deferred.js:84)
    at signalListener (Deferred.js:58)
    at signalWaiting (Deferred.js:28)
    at Deferred.reject.reject (Deferred.js:219)
    at signalDeferred (Deferred.js:84)
    at signalListener (Deferred.js:55) "Error: SyntaxError: Unexpected token < in JSON at position 0
    at http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/NCList.js:181:35
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:37:21)
    at signalWaiting (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:28:4)
    at Deferred.reject.reject (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:219:5)
    at signalDeferred (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:84:15)
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:58:4)
    at signalWaiting (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:28:4)
    at Deferred.reject.reject (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:219:5)
    at signalDeferred (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:84:15)
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:55:5)
    ----------------------------------------
    rejected at signalDeferred (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:84:15)
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:55:5)
    at signalWaiting (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:28:4)
    at Deferred.reject.reject (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:219:5)
    at signalDeferred (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:84:15)
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:58:4)
    at signalWaiting (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:28:4)
    at Deferred.reject.reject (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:219:5)
    at signalDeferred (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:84:15)
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:55:5)
    ----------------------------------------
Error
    at Promise.Deferred.then.promise.then (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:252:24)
    at NCList.<anonymous> (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/NCList.js:171:20)
    at NCList.iterHelper (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/NCList.js:186:16)
    at NCList.iterate (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/NCList.js:221:14)
    at Object._getFeatures (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/SeqFeature/NCList.js:266:29)
    at http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/SeqFeature/NCList.js:242:23
    at signalListener (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:37:21)
    at signalWaiting (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:28:4)
    at Deferred.resolve (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/dojo/Deferred.js:192:5)
    at Object._handleTrackInfo (http://localhost:8098/apollo/18424175391403410330/jbrowse/src/JBrowse/Store/SeqFeature/NCList.js:128:29)"
```

And on the server:
```
| Error 2017-07-18 11:14:03,139 [http-bio-8098-exec-5] ERROR errors.GrailsExceptionResolver  - NullPointerException occurred when processing request: [GET] /apollo/18424175391403410330/jbrowse/data/tracks/Augustus%2012/%7B%22sequenceList%22:[%7B%22name%22:%22Group1.1%22,%22start%22:0,%22end%22:1382403,%22reverse%22:false,%22organism%22:%22Amel%22,%22location%22:[%7B%22fmin%22:0,%22fmax%22:1382403%7D]%7D]%7D:0..1382403/lf-1.json
Stacktrace follows:
Message: null
    Line | Method
->>   32 | findProjectChunkForIndex in org.bbop.apollo.gwt.shared.projection.ProjectionChunkList
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
|   1150 | $tt__projectTrackChunk   in org.bbop.apollo.TrackService
|    349 | data . . . . . . . . . . in org.bbop.apollo.JbrowseController
|    198 | doFilter                 in grails.plugin.cache.web.filter.PageFragmentCachingFilter
|     63 | doFilter . . . . . . . . in grails.plugin.cache.web.filter.AbstractFilter
|    449 | executeChain             in org.apache.shiro.web.servlet.AbstractShiroFilter
|    365 | call . . . . . . . . . . in org.apache.shiro.web.servlet.AbstractShiroFilter$1
|     90 | doCall                   in org.apache.shiro.subject.support.SubjectCallable
|     83 | call . . . . . . . . . . in     ''
|    383 | execute                  in org.apache.shiro.subject.support.DelegatingSubject
|    362 | doFilterInternal . . . . in org.apache.shiro.web.servlet.AbstractShiroFilter
|    125 | doFilter                 in org.apache.shiro.web.servlet.OncePerRequestFilter
|   1142 | runWorker . . . . . . .  in java.util.concurrent.ThreadPoolExecutor
|    617 | run                      in java.util.concurrent.ThreadPoolExecutor$Worker
^    745 | run . . . . . . . . . .  in java.lang.Thread

```

This isn't track specific. Sometimes it shows the features and sometimes it throws this error.

File: client/apollo/js/main.js (hunk 1 above)
Comment (deepakunni3) [inline client/apollo/js/main.js]: Snap! Sorry, I should have kept an eye out for that.

**Comment by nathandunn** [comment]
Yeah, I'll check the Augustus tracks anyway

File: client/apollo/js/main.js (hunk 1 above)
Comment (deepakunni3) [inline client/apollo/js/main.js]: @nathandunn I still get this error after pulling from the projection_v2_server branch.

Here is what I have in my config:
```
jbrowse {
   git {
       url= "https://github.com/GMOD/jbrowse"
       branch = "projection_v2_server"
       alwaysPull = true
       alwaysRecheck = true
   }
```

**Comment by nathandunn** [comment]
@deepakunni3  Try pulling again (and make sure to removed webapps/jbrowse)

I had been pushing into my repo instead of the GMOD one.   You should get a few commits now.

**Comment by deepakunni3** [comment]
@nathandunn Reverse complement works in public mode. Tested for feature tracks. 👍

**Comment by nathandunn** [comment]
Fixed problem with white-space.

**Pull request merged**